src/SlackConnector/SlackConnection.cs

```diff
@@ -87,13 +93,35 @@ public void Initialise(ConnectionInformation connectionInformation)
                 MentionsBot = _mentionDetector.WasBotMentioned(Self.Name, Self.Id, inboundMessage.Text)
             };
 
-            await RaiseMessageReceived(message);
+            return RaiseMessageReceived(message);
+        }
+
+        private Task HandleGroupJoined(GroupJoinedMessage inboundMessage)
+        {
+            var channelId = inboundMessage?.Channel?.Id;
+            if (channelId == null) return Task.FromResult(false);
```

Comment:
yeah, this is what I referred to in the PR

- Channel and Group are two different objects, so its impossible to handle them the same
- I did not want to build a fully fledged "event dispatcher framework" for 3 message types, so I felt that short term this is OK. When more message types are needed this must be refactored something more generic